pkg/kv/kvserver/kvserverpb/raft.proto

```diff
@@ -244,6 +244,12 @@ message SnapshotRequest {
     // keys in key order, as opposed to iterating over point keys first
     // and range keys second. The receiver can take advantage of this
     // to split points/range keys into multiple sstables for ingestion.
+    //
+    // MIGRATION: v24.3 and newer will always set this field to true, and v25.3
+    // and newer stop consulting the field altogether (assume it is true). In
+    // v26.1, we can then drop this field altogether, since v26.1 is the first
```

Comment:
I've been wondering if there is a codified way to express such things and avoid baking assumptions about release policy (which, unlikely, but can change and rot) into the comments. For example, have something like:

```golang
// TODO(#<issue>): when this cluster version is removed, don't set the field,
// because every node is running a later version and assumes such and such.
_ = clusterversion.V25_2
header.RangeKeysInOrder = true
```

I've seen a bunch of comments like this which weren't brought over the finish line because it tends to be forgotten.

Reply:
If it's a useful enough pattern, this could be enforced further:

```golang
if !clusterversion.V25_2.Rotten() {
	// TODO: when removing this branch, also remove RangeKeysInOrder field.
	header.RangeKeysInOrder = true
}
```

When `V25_2` is removed, the `Rotten()` "logically" becomes true, and the branch will be forced to be removed.

Reply:
Yeah, it would be nice to "schedule" the removal at the right time. One issue is that the person doing the MinSupportedVersion bump isn't usually interested in doing a bunch of (sometimes nontrivial) cleanup in order to pass CI.
But I agree - there should be some way. I'll take a look.

Reply:
I'll open another PR to discuss shortly.

Reply:
An opposite extreme would be:

```golang
func TestRemoveRangeKeysInOrder(t *testing.T) {
	if !clusterversion.V25_2.Rotten() {
		t.Skip()
	}
	header := kvserverpb.Header{}
	// TODO: remove this field from proto when this test starts failing.
	// If there are more steps, or the procedure is slightly non-trivial, link
	// to an issue, so that the code owners can be easily notified and help.
	//
	// To break glass, just remove the test altogether, and ping the owners
	// asynchronously.
	_ = header.RangeKeysInOrder
	t.FailNow()
}
```

Reply:
Yes, that's the approach I took, here's the draft: https://github.com/cockroachdb/cockroach/pull/144616/commits/67c7aba94a35ab0c21650c3d1d7922a631dd7c9f